### PR TITLE
by_ethernet now excludes virtual chips

### PR DESCRIPTION
**Pull request by Christian-B**
machine._virtual_chips turned into an OrderedDict to make ii cheaper to check is chip xy is virtual

JsonMachine changed to use the virtual_chips property rather than underscore param

get_existing_xys_by_ethernet updated to check against virtual_chips if needed

get_chips_by_ethernet changed to work based off get_existing_xys_by_ethernet
(only known non test use of get_chips_by_ethernet should be using get_existing_xys_by_etherne anyway.

### CODE REVIEW

File: spinn_machine/full_wrap_machine.py [hunk 1]
```diff
@@ -58,21 +58,21 @@ def get_xy_cores_by_ethernet(self, ethernet_x, ethernet_y):
...
+                           (y + ethernet_y) % self._height)
+                if chip_xy in self._chips and \
+                        chip_xy not in self._virtual_chips:
+                    yield chip_xy
+        else:
+            for (x, y) in self._local_xys:
+                chip_xy = ((x + ethernet_x) % self._width,
+                           (y + ethernet_y) % self._height)
+                if (chip_xy) in self._chips:
+                    yield chip_xy
```
Comment (rowleya) [inline spinn_machine/full_wrap_machine.py:75]: Is it worth having this whole loop twice?  Does the check of an empty virtual_chips take that long?

**Review by rowleya — CHANGES_REQUESTED** [review]
Looks OK in general, but one comment (with repeat as appropriate)

File: spinn_machine/full_wrap_machine.py (hunk 1 above)
Comment (Christian-B) [reply spinn_machine/full_wrap_machine.py:75]: It is a check against an empty OrderedDict so no I dont image it very long.

Was speed vs style choice but don't really mind.

Comment (rowleya) [reply spinn_machine/full_wrap_machine.py:75]: Probably not worth the split in that case :)

**Pull request merged**